### PR TITLE
version documentation on docs site (#1076)

### PR DESCRIPTION
**Pull request by jsilverio22**
This PR adds versioning documentation to the docs website. It includes three make targets to help automate the process of adding, deprecating, and removing versions from the docs and 0.3, 0.4, and 0.5 documentation.

Docs on how to use: https://www.notion.so/5bb0104764434e9aba6b73bf18e2e019?v=46da588d4dfe41808ed80246297930dc
Issue: #1076 
Signed-off-by: Joshua Silverio <joshua@acorn.io>

### CODE REVIEW

**Comment by cjellick** [comment]
Let's discuss this at the end of standup a bit

**Comment by jsilverio22** [comment]
> 

On the 0.4 piece that is not the same behavior I am seeing. When I move to 0.4 tab and click around it seems to be staying within the 0.4 docs. Maybe a specific link is broken that I am missing.

**Comment by jsilverio22** [comment]
> This is a great change! I have two concerns with the current setup.
> 
> First, the docs in v0.4 version link to the latest version. That is, if you are viewing the docs for v0.4 and click a link in the docs, the page that you are taken to is the latest docs. I don't think that is the behavior we want.
> 

I am not seeing this same behavior, when I go to 0.4 and click around it seems to be staying in the 0.4 realm. Maybe a specific link is broken?

**Comment by thedadams** [comment]
Oh, strange! I tried two different links in two different sections to be sure and they both landed me in the `latest` version.

**Comment by jsilverio22** [comment]
oh I see what you mean now, the links within the page, not the sidebar links

**Review by thedadams — COMMENTED** [review]
Just a few questions. I think we'll be good on the next round after we figure out these last few details.

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -29,6 +29,9 @@ const config = {
       /** @type {import('@docusaurus/preset-classic').Options} */
       ({
         docs: {
+          versions: {
+            "0.4": {label: "0.4", banner: "none", path: "0.4"},
+          },
```
[the remark this answers is not in the document]
Comment (jsilverio22) [reply docs/docusaurus.config.js:34]: and this version list is for actively maintained branches besides the latests if 0.5 is added it messes w/ the defaulting

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -51,6 +54,10 @@ const config = {
   themeConfig:
     /** @type {import('@docusaurus/preset-classic').ThemeConfig} */
     ({
+      versions: {
+		"0.5": {label: "0.5", banner: "none", path: "0.5"},
+        "0.4": {label: "0.4", banner: "none", path: "0.4"},
+      },
```
[the remark this answers is not in the document]
Comment (jsilverio22) [reply docs/docusaurus.config.js:60]: Yeah this version list is for the defaulting logic of 0.5 being latest w/o the latest directory

File: Makefile [hunk 3]
```diff
@@ -55,3 +55,47 @@ serve-docs:
...
 	go run tools/gendocs/main.go
+
+#gen new cli docs and copy to specific version
+gen-docs-version:
+	if [ -z ${version} ]; then \
+      			echo "version not set (version=x.x)"; \
+        		exit 1 \
+        	;fi
+	make gen-docs
+	cp -r ./docs/docs/100-reference/01-command-line ./docs/versioned_docs/version-${version}/100-reference
```
[the remark this answers is not in the document]
Comment (jsilverio22) [reply Makefile:66]: yeah I think this was make target was more focused on if the current version we are on needs a patch change then we can use this to help automate that piece. Agreed  probably worth a discussion in standup

File: docs/docusaurus.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (thedadams) [reply docs/docusaurus.config.js:60]: Investigated offline. Not needed.

File: docs/docusaurus.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (thedadams) [reply docs/docusaurus.config.js:34]: Good as-is.

File: Makefile [hunk 4]
```diff
@@ -55,3 +55,47 @@ serve-docs:
...
+  			echo "version not set (version=x.x)"; \
+    		exit 1 \
+    	;fi
+	if [ -z ${prev-version} ]; then \
+  			echo "prev-version not set (prev-version=x.x)"; \
+    		exit 1 \
+    	;fi
+	docker run --rm --workdir=/docs -v $${PWD}/docs:/docs node:18-buster yarn docusaurus docs:version ${version}
+	awk '/versions/&& ++c == 1 {print;print "\t\t\t\"${prev-version}\": {label: \"${prev-version}\", banner: \"none\", path: \"${prev-version}\"},";next}1' ./docs/docusaurus.config.js > tmp.config.js && mv tmp.config.js ./docs/docusaurus.config.js
+	awk '/versions/&& ++c == 2 {print;print "\t\t\"${version}\": {label: \"${version}\", banner: \"none\", path: \"${version}\"},";next}1' ./docs/docusaurus.config.js > tmp.config.js && mv tmp.config.js ./docs/docusaurus.config.js
```
Comment (thedadams) [inline Makefile:80]: Should be removed.

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsilverio22) [reply Makefile]: removed from slack convo

**Review by thedadams — APPROVED** [review]

**Pull request merged**